### ISSUE TITLE
Transient IT failure in GitLab `anvildev` for `develop` pipeline run

### ISSUE THREAD

**Issue by achave11-ucsc** [DataBiosphere/azul#4955]
https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/9838#L1232
```
AIL: test_indexing (integration_test.IndexingIntegrationTest) [other_endpoints] (endpoint=furl('https://indexer.anvil.gi.ucsc.edu'/), path='/health', args=None)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 378, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 535, in _test_other_endpoints
    self._check_endpoint(path, args=args, endpoint=endpoint)
  File "/builds/ucsc/azul/test/integration_test.py", line 630, in _check_endpoint
    return self._get_url_content(url)
  File "/builds/ucsc/azul/test/integration_test.py", line 636, in _get_url_content
    return self._get_url(url).data
  File "/builds/ucsc/azul/test/integration_test.py", line 648, in _get_url
    self._assertResponseStatus(response, expected_statuses)
  File "/builds/ucsc/azul/test/integration_test.py", line 673, in _assertResponseStatus
    assert response.status in expected_statuses, (
AssertionError: (504, 'Gateway Timeout', b'{"message": "Endpoint request timed out"}')
----------------------------------------------------------------------
```

**Comment by achave11-ucsc** [comment]
This was caused by a timeout from `anvildev`'s  indexer /health endpoint.
```
CW Query:
/aws/apigateway/azul-indexer-anvildev
/aws/lambda/azul-indexer-anvildev

fields @timestamp, @message, @logStream, @log
| filter (
    @message like /INTEGRATION_TIMEOUT/ or 
    @requestId like /720d05ad-dad3-4ef8-9bbc-8a16fa00a368/
)
| sort @timestamp desc
| limit 20
```

```
[
    {
        "@timestamp": "2023-02-02 21:57:33.198",
        "@message": "2023-02-02T21:57:33.198Z 720d05ad-dad3-4ef8-9bbc-8a16fa00a368 Task timed out after 31.06 seconds\n\n",
        "@logStream": "2023/02/02/[$LATEST]f7f7761794404f93b583b6900269fcf8",
        "@log": "289950828509:/aws/lambda/azul-indexer-anvildev"
    },
    {
        "@timestamp": "2023-02-02 21:57:33.198",
        "@message": "END RequestId: 720d05ad-dad3-4ef8-9bbc-8a16fa00a368\n",
        "@logStream": "2023/02/02/[$LATEST]f7f7761794404f93b583b6900269fcf8",
        "@log": "289950828509:/aws/lambda/azul-indexer-anvildev"
    },
    {
        "@timestamp": "2023-02-02 21:57:33.198",
        "@message": "REPORT RequestId: 720d05ad-dad3-4ef8-9bbc-8a16fa00a368    Duration: 31060.02 ms    Billed Duration: 31000 ms    Memory Size: 128 MB    Max Memory Used: 118 MB",
        "@logStream": "2023/02/02/[$LATEST]f7f7761794404f93b583b6900269fcf8",
        "@log": "289950828509:/aws/lambda/azul-indexer-anvildev"
    },
    {
        "@timestamp": "2023-02-02 21:57:02.139",
        "@message": "[INFO]	2023-02-02T21:57:02.139Z	720d05ad-dad3-4ef8-9bbc-8a16fa00a368	azul.chalice	Received GET request for '/health', with {"query": null, "headers": {"accept-encoding": "identity", "host": "indexer.anvil.gi.ucsc.edu", "user-agent": "python-urllib3/1.26.12", "x-amzn-cipher-suite": "ECDHE-RSA-AES128-GCM-SHA256", "x-amzn-tls-version": "TLSv1.2", "x-amzn-trace-id": "Root=1-63dc31ae-456d7ef063fb30c46b2573d7", "x-amzn-vpc-id": "vpc-0f09313404dcf5f55", "x-amzn-vpce-config": "1", "x-amzn-vpce-id": "vpce-049c86de06278c666", "x-forwarded-for": "172.73.0.46, 172.73.3.158", "x-forwarded-port": "443", "x-forwarded-proto": "https"}}.",
        "@logStream": "2023/02/02/[$LATEST]f7f7761794404f93b583b6900269fcf8",
        "@log": "289950828509:/aws/lambda/azul-indexer-anvildev"
    },
    {
        "@timestamp": "2023-02-02 21:57:02.139",
        "@message": "[INFO]    2023-02-02T21:57:02.139Z    720d05ad-dad3-4ef8-9bbc-8a16fa00a368    azul.chalice    Did not authenticate request.",
        "@logStream": "2023/02/02/[$LATEST]f7f7761794404f93b583b6900269fcf8",
        "@log": "289950828509:/aws/lambda/azul-indexer-anvildev"
    },
    {
        "@timestamp": "2023-02-02 21:57:02.138",
        "@message": "START RequestId: 720d05ad-dad3-4ef8-9bbc-8a16fa00a368 Version: $LATEST\n",
        "@logStream": "2023/02/02/[$LATEST]f7f7761794404f93b583b6900269fcf8",
        "@log": "289950828509:/aws/lambda/azul-indexer-anvildev"
    },
    {
        "@timestamp": "2023-02-02 21:57:02.102",
        "@message": "{… \"domainName\": \"indexer.anvil.gi.ucsc.edu\", \"domainPrefix\": \"indexer\", \"error_message\": \"Endpoint request timed out\", \"error_messageString\": \" \"Endpoint request timed out\"\", \"error_responseType\": \"INTEGRATION_TIMEOUT\", …, \"stage\": \"anvildev\", \"status\": \"504\"}",
        "@logStream": "1141b712cf430e2be41afb2380961b0e",
        "@log": "289950828509:/aws/apigateway/azul-indexer-anvildev"
    }
]
```

**Comment by achave11-ucsc** [comment]
Update log output with complete Lambda logs.

**Comment by achave11-ucsc** [comment]
@hannes-ucsc: "In order to determine which health check exactly is causing the timeout we would need to add log statements for each check. That can be done by modifying `@health_property`. We should also log any http requests being made during health check. Any usage of `requests` should be replaced with the equivalent usage of `urlib3`."

**Comment by hannes-ucsc** [comment]
For demo (partial, don't close ticket), show new log messages in CloudWatch.

**Comment by achave11-ucsc** [comment]
https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/13423

**Comment by achave11-ucsc** [comment]
https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/13440

**Comment by achave11-ucsc** [comment]
https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/13482#L1421

**Comment by achave11-ucsc** [comment]
@hannes-ucsc: "We haven't seen this since April. We recently improved the performance of the  health check with partial PR #5564 for #5467. We will add boto3 client logging for partial PR #5479 for #5472. We'll follow that up with another PR against #5467 that enables boto3 client logging for S3 and SQS. Hopefully that will enable us to determine  underlying cause of the slow health checks. Once we did that, we will decide if there's anything else to be done here. If not, this ticket will likely become a freebie."
